### ISSUE TITLE
Test Ledger Integration against Cosmos SDK 0.39 signing changes

### ISSUE THREAD

**Issue by zmanian** [cosmos/ledger-cosmos-go#13]
Test ledger integrations with the effects of

https://github.com/cosmos/cosmos-sdk/issues/6213

**Comment by jleni** [comment]
what is the timeframe/deadline of this? planned release date?

**Comment by ebuchman** [comment]
The intention is to start testing this on testnets this summer (~late July/early August), and ideally for it to go live on the hub sometime in the fall.

**Comment by jleni** [comment]
I will put it in my calendar, so we can catch up at the last week of July

**Comment by jleni** [comment]
@ebuchman @zmanian How are you progressing?
Are we at a point where i should start looking at this and check the code/specs on your side?

**Comment by webmaster128** [comment]
@zmanian can you adapt 0.39 -> 0.40 in the title of this ticket to avoid confusion? The now released Cosmos SDK 0.39 (Launchpad) does not change signing.

**Comment by ebuchman** [comment]
I believe full integration is still a work in progress (cc @alexanderbez @aaronc) but there is initial code for the protobuf based txs and unit tests. It would probably be helpful if you'd review the [ADR-20](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-020-protobuf-transaction-encoding.md) if you haven't yet which describes how this is going to look.

**Comment by aaronc** [comment]
You can enable protobuf transactions in tests and simd with the `test_proto` flag. We're still fixing the remaining failing unit tests but most stuff works.

**Comment by zmanian** [comment]
We expect that the initial target for the Ledger app will be ensuring the SIGN_MODE_LEGACY_AMINO sign mode. Now that all the changes are merged into Cosmos SDK master we will be assessing any impactful changes to the Legacy Amino.

I'm unclear on whether or not we want to consider supporting SIGN_MODE_DIRECT in the Ledger. My initial instinct would be not support it.

Eventually we will want to support SIGN_MODE_TEXTUAL.

**Comment by marbar3778** [comment]
closing this for now since ledger works with 0.45. Once sign mode textual lands we would need to come back here
